.github/workflows/test.yaml

```diff
@@ -47,18 +41,31 @@ jobs:
 
       - name: Install dependencies
         run: |
-          python -m pip install -e .[develop,dash]
+          python -m pip install --upgrade -e .[develop,dash]
```

Comment:
so that we test a "fresh" pull, as some dependencies might break even with minor changes, we want to catch that before it affects a user

Reply:
Are you not concerned about dependency breakage? We should then lock our dependencies into a `requirements.txt` or `constraints.txt` and also use that for development. With `--upgrade` here we are always on the latest version of any library on the CI, which 1. makes caching the python environment redundant, 2. means we have a non-reproducible environment.

Reply:
I like to develop on bleeding edge solving issues as they arise, but for version releases I think we should lock our dependencies (and the transient dependencies, if possible).

Reply:
the caching is definitely not redundant, as many packages dont get a new version every week, meaning that `pip` will be faster regardless

Reply:
The caching operates on a much lower level than the pip cache. Being on the bleeding edge also means making choices on which versions we support. I'd rather we support a large number of versions a be a bit 'behind' than to be on the bleeding edge and keep chasing small changes.

With the strategy in this PR, we may even end up with different versions between different runs of the CI. For example, numpy, scipy, and xarray have all dropped support for python 3.7 earlier this year, which means we cannot be on the bleeding edge even if we wanted to. We should pin those as dependencies to test for compatibility.

Reply:
Sure, I agree, this is a big problem with Python anyway.

Then we should put (as much as possible) version numbers in our setup.cfg, and don't update it too often (because that will refresh the github ci cache)

Reply:
I will remove the `--upgrade`